src/ast.cc

```diff
@@ -63,7 +63,8 @@ AST::AST()
     backtrack_product(0),
     backtrack_filter(0),
     original_product(0),
-    char_type(0) {
+    char_type(0),
+    checkpoint(0) {
```

Comment:
In C++, `NULL` or `nullptr` are essentially just macros/typedefs that eventually expand to `0`, so `0` can also be used whenever you want to initialize a ptr as `NULL` or check if a pointer == `NULL`. But I guess it would be a little more clear that a ptr is initialized here if I change it to `nullptr`, so I will change that.